target/linux/ramips/image/mt7621.mk

```diff
@@ -631,12 +637,15 @@ define Device/netis_wf2881
   KERNEL_SIZE := 4096k
   IMAGE_SIZE := 129280k
   UBINIZE_OPTS := -E 5
+  DEVICE_VENDOR := NETIS
+  DEVICE_MODEL := WF2881
+  UIMAGE_NAME := WF2881_0.0.00
+  KERNEL_INITRAMFS := $(KERNEL_DTB) | \
+	netis-tail $$(DEVICE_MODEL) $$(UIMAGE_NAME) | uImage lzma
```

Comment:
Is it expected that the first argument here will always match DEVICE_MODEL or is this a coincidence?
In case of the latter, I'd consider putting "WF2881" directly into the argument list here.

Reply:
Similar (yet different) argument applies for UIMAGE_NAME: If the second argument is always UIMAGE_NAME, consider using that variable in netis-tail directly? If not, keep the current version.

Reply:
The second argument must match UIMAGE_NAME, that's for sure. But how do we know the first one is a coincidence or not? I checked three netis device FW and at least their first arguments matched their model names.

Reply:
> I checked three netis device FW and at least their first arguments matched their "model names".

Okay, so it's still a parameter set by the vendor FW. So it might be possible that we will choose a name different from that in the future (like with Netgear where different device have R6950 for model name in vendor FW, but "our" device name is chosen according to the marketed name; e.g. [1]).

So, I'd use something like the following (just quickly written, haven't checked correct number of $ etc.):
```
define Build/netis-tail
	echo -n "$(1)" >> $@
	echo -n "$$(UIMAGE_NAME)-yun" | $(STAGING_DIR_HOST)/bin/mkhash md5 | \
		sed 's/../\\\\x&/g' | xargs echo -ne >> $@
endef
```

```
  KERNEL_INITRAMFS := $(KERNEL_DTB) | \
	netis-tail WF2881 | uImage lzma
```

One could even go further and introduce a new variable for the device name in FW, like SERCOMM_HWNAME for Netgear. But maybe that's not really necessary after all.

What do you think?

[1] https://github.com/openwrt/openwrt/pull/2614/commits/0ead25fd570c8919b6b8a0c953f2cc7808367a2f#diff-5476a825952c25a3c92968c68aeabce7R595

Reply:
I tried using DEVICE_MODEL or UIMAGE_NAME directly in netis-tail and it failed. `echo -n $$(UIMAGE_NAME)-yun` prints `-yun`.

Reply:
This $ vs. $$ is really hard ...

Reply:
> What do you think?

I guess you're right. I'll follow your advice, thanks.